### PR TITLE
[lexical] Bug Fix: restore managed line break in wrapped slot hosts

### LINKED ISSUE #9063: Bug: Reconciler does not insert a managed line break into ElementDOMSlot
Lexical version: 0.49 latest

## Steps To Reproduce

1. Define an ElementNode with the `getDOMSlot` method
2. Insert a node with text content into the editor 
3. Delete the text content

Link to code example: https://codesandbox.io/p/sandbox/3xpqzc

## The current behavior

When a node with a DOM slot is empty, it lacks a managed line break, so node visually collapses as if it had been completely removed

https://github.com/user-attachments/assets/c6ff08fe-f1c3-4dca-86c1-c4fd0acfce1e

## The expected behavior

A managed line break is set if the DOMSlot is empty, similar to the standard DOM structure for an ElementNode

## Impact of fix

It would be better if the managed line break were handled entirely by lexical

### PR DESCRIPTION
**Pull request by luantaraschi**
## Description

An element with named slots can redirect its linked-list children to a separate DOM element through `getDOMSlot().withElement(...)`. When that child area becomes empty, it still needs a managed line break even though the host has content in a named slot.

`ElementDOMSlot` now owns this decision. Named-slot containers form a contiguous prefix of its managed range, so it only checks the first node after the `after` boundary, or the element's first child when there is no boundary. A wrapped child area therefore receives the line break, while the default slot keeps the existing slots-only behavior. The reconciler delegates the decision on each relevant render, so adding or removing a slot updates the managed line break without leaving stale cached state.

Closes #9063

### Before

The wrapped-host regression fails because the redirected content element has no managed line break. The boundary regression also fails because a slot container before the managed range incorrectly suppresses the line break.

### After

The wrapped content element receives the managed line break. Removing the named slot from a childless default host restores its line break, and adding the slot again removes it. Slot containers before an `after` boundary do not affect the managed range.

- Core unit suite: 56 files, 1,269 tests passed.
- Browser suite: 51 files, 365 tests passed and 4 skipped across Chromium, Firefox, and WebKit.
- TypeScript, ESLint, Prettier, `git diff --check`, and the pre-commit hook passed.

### CODE REVIEW

**Review by etrepum — COMMENTED** [review]
I think the right solution here is to have the default ElementDOMSlot implementation slot-aware

**Comment by luantaraschi** [comment]
Thanks, that direction makes sense. I moved the named-slot handling into `ElementDOMSlot`. It now decides whether an empty managed line break is needed from the slot containers in its own element, while the reconciler only reports the linked-list last-child state.

I also added coverage for removing and adding a slot so the cached state cannot become stale. The full core unit suite passed with 1,268 tests. The browser suite passed with 263 tests across Chromium, Firefox, and WebKit, with 4 existing skips. TypeScript, ESLint, Prettier, `git diff --check`, and the pre-commit hook also passed.

File: packages/lexical/src/LexicalDOMSlot.ts [hunk 1]
```diff
@@ -431,11 +431,19 @@ export class ElementDOMSlot<
     lineBreakType: null | 'empty' | 'line-break' | 'decorator',
   ): void {
     const element: HTMLElement & LexicalPrivateDOM = this.element;
-    element.__lexicalLastChildKind = lineBreakType;
-    if (lineBreakType === null) {
+    const nextLineBreakType =
+      lineBreakType === 'empty' &&
+      Array.from(element.children).some(isSlotContainerDOM)
```
Comment (etrepum) [inline packages/lexical/src/LexicalDOMSlot.ts]: This seems suspicious, I don't think that every child should have to be inspected

Comment (luantaraschi) [reply packages/lexical/src/LexicalDOMSlot.ts]: You're right, scanning every child wasn't necessary. Slot containers form a contiguous prefix of the managed range, so this now checks only the first node after the `after` boundary, or `element.firstChild` when there is no boundary. That makes the check constant-time and avoids counting slot containers outside the managed range.

I added a regression for the boundary case. It fails with the previous scan and passes with this change. The full core unit suite passed with 1,269 tests. The browser suite passed with 365 tests across Chromium, Firefox, and WebKit, with 4 skips. TypeScript, ESLint, Prettier, `git diff --check`, and the pre-commit hook also passed.

**Review by etrepum — APPROVED** [review]

**Review by etrepum — APPROVED** [review]
